staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -91,18 +92,30 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	result := make(chan interface{})
+	type result struct {
+		err   interface{}
+		stack string
+	}
+	resCh := make(chan result)
 	tw := newTimeoutWriter(w)
 	go func() {
 		defer func() {
-			result <- recover()
+			res := result{
+				err: recover(),
+			}
+			if res.err != nil {
+				buf := make([]byte, 2048)
+				n := runtime.Stack(buf, false)
```

Comment:
the corresponding [http.Server](https://golang.org/src/net/http/server.go#L1744) and [http2.Server](https://github.com/kubernetes/kubernetes/blob/c66fa57476ae441e8dded9149934b4141097499c/vendor/golang.org/x/net/http2/server.go#L2056-L2058) impls use a bigger buffer than this... any reason not to preserve more?

Reply:
No specific reason other than it was the size in the runtime package tests.

Updated.